### PR TITLE
added missing and conditional completions

### PR DESCRIPTION
**Pull request by tbhaxor**
## Description

Improved the completions and added missing argument's completions

## Motivation and Context

Completion for wpscan tool

## How Has This Been Tested?

Manual

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [x] I have added tests to cover my changes, and all the new and existing tests pass.

### CODE REVIEW

**Comment by NoahGorny** [comment]
Hi @tbhaxor 
Nice work :smile:

Can you please split this PR into two separate cleanups? one for vuejs and one for wpscan?

**Comment by tbhaxor** [comment]
Sure give me some time @NoahGorny :P I work on open source projects in the week ends

File: completion/available/wpscan.completion.bash [hunk 1]
```diff
@@ -1,15 +1,62 @@
...
-        for _opt_ in ${OPTS[@]}; do
-            if [[ "$_opt_" == "$2"* ]]; then
-                COMPREPLY+=("$_opt_")
-            fi
-        done
-    }
-
-    complete -F __wpscan_completion wpscan
+
+	_wpscan()   {
```
Comment (cornfeedhobo) [inline completion/available/wpscan.completion.bash:5]: It's a common convention to name completion functions with __<cmd name>_<function purpose>.

Could you preserve `__wpscan_completion`?

**Review by cornfeedhobo — CHANGES_REQUESTED** [review]
@tbhaxor Thanks as always for your contributions! I've requested a change, but I'm also going to mark this a **blocked** by #1760, because you'll notice that your files are using inconsistent white space due to a typo in our `.editorconfig` (cc @NoahGorny).

**Comment by cornfeedhobo** [comment]
@tbhaxor #1760 has been merged. If you would rebase this branch to master and then re-format your files, I'll take another look. Thanks again!

**Pull request closed without merging**